src/mito2/src/region/version.rs

```diff
@@ -108,6 +108,23 @@ impl VersionControl {
         data.is_dropped = true;
         data.version.ssts.mark_all_deleted();
     }
+
+    /// Reset current version.
+    pub(crate) fn reset(&self, flushed_entry_id: u64, memtable_builder: &MemtableBuilderRef) {
+        let version = self.current().version;
+
+        let new_mutable = memtable_builder.build(&version.metadata);
+        let new_memtables = MemtableVersion::new(new_mutable);
+
+        let mut version_builder = VersionBuilder::from_version(version).memtables(new_memtables);
+        version_builder.flushed_entry_id = flushed_entry_id;
+        version_builder.ssts = Arc::new(SstVersion::new());
+        let new_version = Arc::new(version_builder.build());
```

Comment:
suggestion: Could we add a setter `fn flushed_entry_id(mut self, id: EntryId) -> VersionBuilder`?
```suggestion
        let new_version = Arc::new(VersionBuilder::new(version.metadata.clone(), new_mutable).flushed_entry_id(version.flushed_entry_id).build());
```